dependent_type_theory.md

```diff
@@ -251,7 +264,7 @@ similarly polymorphic:
 #check Prod    -- Type u_1 → Type u_2 → Type (max u_1 u_2)
 ```
 
-To define polymorphic constants, Lean allows you to
+To define polymorphic functions, Lean allows you to
```

Comment:
We should keep "constants". Recall that we can define universe polymorphic constants that are not functions in Lean. 
Let's add "universe" here to make clear this paragraph is about universe polymorphism.
"""
To define universe polymorphic constants, ...
"""

Reply:
Well the problem is we've removed most "constant" examples because they were confusing, so adding it back here is also confusing, how about we show how to define a polymorphic function, then add after that that in fact you can define any polymorphic constant, which can be functions, or values, or ??? (what else is a 'constant' in lean?)

Reply:
We have removed the `constant` command examples to avoid confusion such as: only symbols declared with the `constant` keyword are constants. `Prod` is a constant in Lean, but it was not declared with the `constant` command.

Reply:
> what else is a 'constant' in lean?

Every declaration in the environment is a constant.
The environment is basically a mapping from constant names to their data (type, universe parameters, etc). There are different kinds. The complete list is here:
https://github.com/leanprover/lean4/blob/master/src/Lean/Declaration.lean#L309-L318

Reply:
@leodemoura In my opinion, this is another good reason to rename the `constant` command to something more apt (e.g., `opaque`) to avoid such confusion.

Reply:
@tydeu Yes, I agree. Sebastian and I were just talking about that. We want to add a "deprecated warning" message at `constant`, and move to `opaque` (or something similar)

Reply:
We also decided to replace the word "constant" with "declaration" in most places in the documentation. It works in most cases, and it will be easier to understand.

Reply:
great, can I use declaration here then too?

Reply:
2 thumbs up so I went with "declaration" so I could resolve this conversation.